### PR TITLE
update analytics context in analytics service when started

### PR DESCRIPTION
**Pull request by sohanirao**

### CODE REVIEW

File: src/runtime/analytics-service.js [hunk 1]
```diff
@@ -84,21 +97,63 @@ export class AnalyticsService {
     return this.iframe_;
   }
 
+  /**
+   * @return {!string}
+   * @private
+   */
+  getQueryString_() {
+    return this.doc_.getWin().location.search.split('?')[1];
```
Comment (dvoytenko) [inline src/runtime/analytics-service.js]: IIRC, you don't need to remove `?`.

File: src/runtime/analytics-service.js [hunk 2]
```diff
@@ -84,21 +97,63 @@ export class AnalyticsService {
     return this.iframe_;
   }
 
+  /**
+   * @return {!string}
```
Comment (dvoytenko) [inline src/runtime/analytics-service.js]: `!string` -> `string`. True for all primitives in JavaScript.

File: src/runtime/analytics-service.js [hunk 3]
```diff
@@ -84,21 +97,63 @@ export class AnalyticsService {
...
+    return this.doc_.getWin().document.referrer;
+  }
+
+  /**
+   * @return {!Promise}
+   */
+  setContext_() {
+    const utmParams = parseQueryString(this.getQueryString_());
+    this.context_.setTransactionId(this.xid_.get());
+    this.context_.setReferringOrigin(this.getReferrer_());
```
Comment (dvoytenko) [inline src/runtime/analytics-service.js]: Wrap it in `parseUrl().origin` or `getOrigin()`

File: src/runtime/analytics-service.js [hunk 4]
```diff
@@ -84,21 +97,63 @@ export class AnalyticsService {
...
           this.portResolver_(port);
+          return this.setContext_();
         });
   }
 
   /**
    */
   close() {
+    // Clean up context
+    this.context_ = new AnalyticsContext();
```
Comment (dvoytenko) [inline src/runtime/analytics-service.js]: It's defined as `@const` up top.

File: src/runtime/transaction-id.js [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+    /** @private @const {!Storage} */
+    this.storage_ = deps.storage();
+  }
+
+  /**
+   * Returns the current transaction id
+   *  @return {!Promise<string>}
+   */
+  get() {
+    return this.storage_.get('transaction_id').then(id => {
```
Comment (dvoytenko) [inline src/runtime/transaction-id.js]: I believe we decided to drop storage already? Why do we still need it?

Comment (sohanirao) [reply src/runtime/transaction-id.js]: Based on offline discussion, we will keep this for now.

File: src/runtime/analytics-service.js (hunk 4 above)
Comment (sohanirao) [reply src/runtime/analytics-service.js]: will replace with todo for now.

File: src/runtime/analytics-service.js (hunk 1 above)
Comment (sohanirao) [reply src/runtime/analytics-service.js]: Done

File: src/runtime/analytics-service.js [hunk 6]
```diff
@@ -19,25 +19,27 @@ import {createElement} from '../utils/dom';
...
   display: 'none',
 };
 
+
 export class AnalyticsService {
   /**
-   * @param {!../model/doc.Doc} doc
-   * @param {!web-activities/activity-ports.ActivityPorts} activityPorts
-   * @param {!../model/page-config.PageConfig} config
+   * @param {!./deps.DepsDef} deps
```
Comment (chenshay) [inline src/runtime/analytics-service.js]: The naming is confusing for me. deps isn't very descriptive. When instantiating this you're passing a ConfiguredRuntime type.

Comment (sohanirao) [reply src/runtime/analytics-service.js]: Runtime implements those APIs, and this is the pattern we follow for all flows.

File: src/runtime/analytics-service.js (hunk 3 above)
Comment (sohanirao) [reply src/runtime/analytics-service.js]: done

File: src/runtime/analytics-service.js (hunk 6 above)
Comment (dvoytenko) [reply src/runtime/analytics-service.js]: `ConfiguredRuntime` implements `Deps`, which is short for dependencies. But it's not configured runtime itself.

File: src/runtime/analytics-service.js [hunk 7]
```diff
@@ -84,21 +96,62 @@ export class AnalyticsService {
...
+  getQueryString_() {
+    return this.doc_.getWin().location.search.slice(1);
+  }
+
+  /**
+   * @return {string}
+   * @private
+   */
+  getReferrer_() {
+    return parseUrl(this.doc_.getWin().document.referrer).origin;
```
Comment (dvoytenko) [inline src/runtime/analytics-service.js]: I'd rather have `parse().origin` to be done outside of this class. This is something that we critically need to test and `getReferrer_` is here to mock out this part of code.

File: src/runtime/analytics-service.js [hunk 8]
```diff
@@ -84,21 +96,62 @@ export class AnalyticsService {
     return this.iframe_;
   }
 
+  /**
+   * @return {string}
+   * @private
+   */
+  getQueryString_() {
+    return this.doc_.getWin().location.search.slice(1);
```
Comment (dvoytenko) [inline src/runtime/analytics-service.js]: No need for `slice(1)`

File: src/runtime/analytics-service.js [hunk 9]
```diff
@@ -84,21 +96,62 @@ export class AnalyticsService {
...
         this.args_).then(port => {
           this.portResolver_(port);
+          return this.setContext_();
         });
   }
 
   /**
    */
   close() {
+    // TODO(sohanirao): Clean up context?
```
Comment (dvoytenko) [inline src/runtime/analytics-service.js]: No need in this `TODO`. I think we can keep it safely dirty.

File: src/runtime/analytics-service.js (hunk 7 above)
Comment (sohanirao) [reply src/runtime/analytics-service.js]: done

File: src/runtime/analytics-service.js (hunk 8 above)
Comment (sohanirao) [reply src/runtime/analytics-service.js]: done

File: src/runtime/analytics-service.js (hunk 9 above)
Comment (sohanirao) [reply src/runtime/analytics-service.js]: done

**Pull request merged**